app/models/planning_application.rb

```diff
@@ -1133,16 +1133,19 @@ def audit_update_application_type!
     old_application_type = ApplicationType.find(changes["application_type_id"].first)
     old_reference = reference
 
-    set_application_number
-    set_reference
-
-    audit!(
-      activity_type: "updated",
-      activity_information: "Application type",
-      audit_comment:
-        "Application type changed from: #{old_application_type.full_name} / Changed to: #{application_type.full_name},
+    transaction do
+      set_application_number
```

Comment:
I think if we call this method only when the application number is nil, we'll also gain the behaviour of reusing the existing application number when the type changes. I know that wasn't a vital requirement, though.

Reply:
It's only triggered with `before_update :audit_update_application_type!, if: :application_type_id_changed?`
so perhaps not needed at all?

Reply:
Yeah, I think it should only ever need to be set once anyway. If a new application gets a number then that should remain valid forever.